commons/src/test/java/com/powsybl/commons/report/ReportNodeTest.java

```diff
@@ -336,62 +335,149 @@ void testTimestamps() {
 
     @Test
     void testMissingKey() {
-        // Without giving a locale
+        // Default locale for this test class is Locale.US
         ReportNode report1 = ReportNode.newRootReportNode()
-                .withResourceBundles(PowsyblCoreTestReportResourceBundle.TEST_BASE_NAME, PowsyblCoreReportResourceBundle.BASE_NAME)
+                .withResourceBundles(TEST_BASE_NAME)
                 .withMessageTemplate("unknown.key")
                 .build();
-        // translation should fall back to default properties as the key is not defined in the reports_en_US.properties
+        // translation should fall back to the default powsybl core reports.properties 'core.commons.missingKey' template because no reports_en.properties or reports_en_US.properties exist
         assertEquals("Cannot find message template with key: 'unknown.key'", report1.getMessage());
 
         // With Locale.FRENCH
         ReportNode report2 = ReportNode.newRootReportNode()
-                .withResourceBundles(PowsyblCoreTestReportResourceBundle.TEST_BASE_NAME, PowsyblCoreReportResourceBundle.BASE_NAME)
                 .withLocale(Locale.FRENCH)
+                .withResourceBundles(TEST_BASE_NAME)
                 .withMessageTemplate("unknown.key")
                 .build();
-        // translation should fall back to default properties as the key is not defined in the reports_en_US.properties
+        // translation should fall back to the powsybl core reports_fr.properties 'core.commons.missingKey' template because it exists and the report node root locale is FRENCH
         assertEquals("Template de message non trouvé pour la clé 'unknown.key'", report2.getMessage());
     }
 
     @Test
-    void testLocaleAndi18n() {
-        // Without giving a locale => default one is en_US
-        ReportNode rootReportEnglish = ReportNode.newRootReportNode()
-                .withResourceBundles(PowsyblCoreTestReportResourceBundle.TEST_BASE_NAME, PowsyblCoreReportResourceBundle.BASE_NAME)
-                .withMessageTemplateProvider(new BundleMessageTemplateProvider(TEST_BASE_NAME))
-                .withMessageTemplate("rootWithValue")
-                .withUntypedValue("value", 4)
-                .build();
-        // translation should fall back to default properties as the key is not defined in the reports_en_US.properties
-        assertEquals("Root message with value 4", rootReportEnglish.getMessage());
-        assertEquals(Locale.US, rootReportEnglish.getTreeContext().getLocale());
+    void testNoJVMDefaultFallbackWithLocaleExistingReportButNoTemplateForKey() {
+        // Set Locale default as French to be the fallback if English report is not found
+        // as the English report exists for tests, then the fallback in case the key is not found in the English report
+        // should not be in French but in the default reports.properties
+        Locale previousLocale = Locale.getDefault();
+        try {
+            Locale.setDefault(Locale.FRENCH);
```

Comment:
The tests may interfere with each other, and with ReportConstantsTest::testGetDefaultReportLocale, when run in parallel. Coulnd't we rather mock `ReportConstants.getDefaultLocale()` to avoid this?